x-pack/plugins/files/server/routes/file_kind/create.ts

```diff
@@ -26,13 +26,18 @@ const rt = {
 export type Endpoint<M = unknown> = CreateRouteDefinition<typeof rt, { file: FileJSON<M> }>;
 
 export const handler: CreateHandler<Endpoint> = async ({ fileKind, files }, req, res) => {
-  const { fileService } = await files;
+  const { fileService, security } = await files;
   const {
     body: { name, alt, meta, mimeType },
   } = req;
-  const file = await fileService
-    .asCurrentUser()
-    .create({ fileKind, name, alt, meta, mime: mimeType });
+  const file = await fileService.asCurrentUser().create({
+    fileKind,
+    name,
+    alt,
+    meta,
+    mime: mimeType,
+    owner: security?.authc.getCurrentUser(req)?.username,
```

Comment:
I am not 100% sure that just `username` is enough to identify the user, as I saw this in the search code:

https://github.com/elastic/kibana/blob/7a3243b79fbc309875490e1477a3d97b80854676/src/plugins/data/common/search/session/types.ts#L53-L58


I'd ask @elastic/kibana-security for a review if we haven't already

Reply:
This is a good point @Dosant . We reached out to security team and they advised us to use `profile_uid`. Changes incoming.